sound/soc/intel/boards/sof_ssp_amp.c

```diff
@@ -499,6 +238,4 @@ MODULE_DESCRIPTION("ASoC Intel(R) SOF Amplifier Machine driver");
 MODULE_AUTHOR("Balamurugan C <balamurugan.c@intel.com>");
 MODULE_AUTHOR("Brent Lu <brent.lu@intel.com>");
 MODULE_LICENSE("GPL");
-MODULE_IMPORT_NS(SND_SOC_INTEL_HDA_DSP_COMMON);
-MODULE_IMPORT_NS(SND_SOC_INTEL_SOF_REALTEK_COMMON);
-MODULE_IMPORT_NS(SND_SOC_INTEL_SOF_CIRRUS_COMMON);
+MODULE_IMPORT_NS(SND_SOC_INTEL_SOF_SSP_COMMON);
```

Comment:
common comment for all machine changes: this PR is a bit too invasive. 

The notion of detecting codecs with ACPI is fine, it's the right thing to do. Can we start with this first? 

I don't see the point of trying to change DMIC and HDMI support at the same time, and adding common parts to model different dailink orders. This is starting to bring limited returns compared to lots of code changes. 

Can we start with the codec detection first, and limit the PR to this? It'll be a lot easier to review and merge.